### PR TITLE
[receiver/snowflakereceiver] added more detailed readme

### PR DESCRIPTION
**Pull request by shalper2**
**Description:** added README in preparation of graduating component to alpha

**Link to tracking Issue:** [14754](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14754)

**Documentation:** Documentation was added to README.md in component root.

### CODE REVIEW

File: receiver/snowflakereceiver/testdata/config.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  account: bigbusinessaccount
+  warehouse: metricWarehouse
+  # optional settings (i.e. have defaults)
+  collection_interval: 18m
+  metrics:
+    snowflake.database.bytes_scanned.avg:
+      enabled: true
+    snowflake.database.bytes_deketed.avg:
+      enabled: false
+  role: customMonitoringRole
```
Comment (dmitryax) [inline receiver/snowflakereceiver/testdata/config.yaml:14]: These config files in testdata are usually used in the tests like here. Feel free to add a test for this config in a follow-up PR

**Review by dmitryax — APPROVED** [review]

File: receiver/snowflakereceiver/README.md [hunk 2]
```diff
@@ -6,7 +6,39 @@
 | Supported pipeline types | metrics       |
 | Distributions            | [contrib]     |
 
-This receiver collects metrics from a Snowflake account by connecting to an account and running queries at set intervals.
+This receiver collects metrics from a Snowflake account by connecting to and querying a Snowflake deployment.
 
-[development]: https://github.com/open-telemetry/opentelemetry-collector#development
-[contrib]:https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib
```
Comment (dmitryax) [inline receiver/snowflakereceiver/README.md:12]: Please don't remove the anchor links. they are used in the table above

File: receiver/snowflakereceiver/testdata/config.yaml (hunk 1 above)
Comment (shalper2) [reply receiver/snowflakereceiver/testdata/config.yaml:14]: sure no problem, probably should have to begin with

**Review by codeboten — COMMENTED** [review]
Please rebase to pick up the changes to run go1.20 unit tests in CI

**Pull request merged**